Gems/Atom/Feature/Common/Code/Source/FrameCaptureSystemComponent.cpp

```diff
@@ -509,50 +496,54 @@ namespace AZ
             return InvalidFrameCaptureId;
         }
 
+        FrameCaptureId FrameCaptureSystemComponent::InternalCaptureScreenshot(
+            const AZStd::string& imagePath, AzFramework::NativeWindowHandle windowHandle)
+        {
+            CaptureHandle captureHandle = ScreenshotPreparation(imagePath, nullptr);
+            if (captureHandle.IsNull())
+            {
+                return InvalidFrameCaptureId;
+            }
+            AZStd::scoped_lock<CaptureHandle> scope_lock(captureHandle);
+            CaptureState* captureState = captureHandle.GetCaptureState();
+
+            // Find SwapChainPass for the window handle
+            RPI::SwapChainPass* pass = AZ::RPI::PassSystemInterface::Get()->FindSwapChainPass(windowHandle);
+            if (!pass)
```

Comment:
This check could be done before the call to ScreenshotPreparation, would make the logic easier as you don't have to return a capture back to idle.

Reply:
Good point. Swapped the order.